app/_src/gateway/kong-enterprise/fips/fips-support.md

```diff
@@ -0,0 +1,54 @@
+---
+title: About FIPS 140-2 Compliance in Kong Gateway
+badge: enterprise
+content_type: reference
+---
+
+The Federal Information Processing Standard (FIPS) 140-2 is a federal standard defined by the National Institute of Standards and Technology. It specifies the security requirements that must be satisfied by a cryptographic module. The FIPS {{site.base_gateway}} package is FIPS 140-2 compliant. Compliance means that the software has met all of the rules of FIPS 140-2, but has not been submitted to a NIST testing lab for validation.
+
+
+{{site.ee_product_name}} provides a FIPS 140-2 compliant package for **Ubuntu 20.04** {% if_version gte:3.1.x %}, **Ubuntu 22.04**, and **Red Hat Enterprise 8** {% endif_version %}. This package provides compliance for the core {{site.base_gateway}} product{% if_version gte:3.1.x %} and all out of the box plugins{% endif_version %}.
+
+The package replaces the primary library in {{site.base_gateway}}, OpenSSL, with [BoringSSL](https://boringssl.googlesource.com/boringssl/), which at its core uses the FIPS 140-2 validated BoringCrypto for cryptographic operations.
+
+## FIPS implementation
+### Password hashing
+
+Password hashing is designed to protect the original password from being reversed. Although FIPS 140-2 doesn't mention password hashing, {{site.base_gateway}} uses key derivation functions that are approved:
+
+| Component | Normal mode | FIPS mode | Notes |
+|-----------|-------------|-----------|-------|
+| core/rbac | bcrypt | PBKDF2 [^1] | PBKDF2 in BoringSSL isn't FIPS validated. OpenSSL 3.0 is required. |
+| plugins/oauth2[^2][^3] | Argon2 or bcrypt (when `hash_secret=true`) | Disabled (`hash_secret` can’t be set to `true`) | PBKDF2 in BoringSSL isn't FIPS validated. OpenSSL 3.0 is required. |
+
+[^1]: As of {{site.base_gateway}} FIPS 3.0, RBAC uses PBKDF2 as password hashing algorithm. Although PBKDF2 is an approved algorithm in FIPS 140-2, the BoringSSL implementation of PBKDF2 is not validated. This approach is future proof because once OpenSSL 3.0 gets its FIPS certification, we can move off BoringSSL, and OpenSSL 3.0’s PBKDF2 implementation is included in its FIPS validation.
+[^2]: As of {{site.base_gateway}} FIPS 3.1, the oauth2 plugin disables `hash_secret` feature, so the user can’t turn it on. This means password will be stored plaintext in the database; however, users can choose to use secrets management or db encryption instead.
+[^3]: As of {{site.base_gateway}} FIPS 3.1, key-auth-enc uses SHA1 to speed up lookup of a key in DB. As of {{site.base_gateway}} FIPS 3.2, SHA1 support is “read-only”, meaning existing credentials in DB are still validated, but any new credentials will be hashed in SHA256.
```

Comment:
@raoxiaoyan We don't mention key-auth-enc specifically in the table above, is this included in `plugins/oauth2`?

Reply:
These are questions for @fffonion, my fault for citing wrong SME on source ticket! my apologies.

Reply:
We should add key-auth-enc to the table above yes, they are two different plugins. That was my fault : )